docs/hub/datasets-gated.md

```diff
@@ -84,17 +84,33 @@ By default, users landing on your gated dataset will be asked to share their con
     <img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/datasets-gated-user-side-dark.png"/>
 </div>
 
-If you want to collect more user information, you can configure additional fields. This information will be accessible from the **Settings** tab. To do so, add an `extra_gated_fields` property to your [dataset card metadata](./datasets-cards#dataset-card-metadata) containing a list of key/value pairs. The *key* is the name of the field and *value* its type. A field can be either `text` (free text area) or `checkbox`. Finally, you can also personalize the message displayed to the user with the `extra_gated_prompt` extra field.
+If you want to collect more user information, you can configure additional fields. This information will be accessible from the **Settings** tab. To do so, add an `extra_gated_fields` property to your [dataset card metadata](./datasets-cards#dataset-card-metadata) containing a list of key/value pairs. The *key* is the name of the field and *value* its type or an object with a `type` field. The list of field types is:
+
+- `text`: a single-line text field.
+- `checkbox`: a checkbox field.
+- `date_picker`: a date picker field.
+- `country`: a country dropdown. The list of countries is based on the [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) standard.
+- `select`: a dropdown with a list of options. The list of options is defined in the `options` field. Example: `options: ["option 1", "option 2", {label: "option3", value: "opt3"}]`.
+
+Finally, you can also personalize the message displayed to the user with the `extra_gated_prompt` extra field.
```

Comment:
I would be opinionated and advertise only the following format in the docs (object with type)
```yaml
Country:
   type: "country"
"Date of Birth":
   type: "date_picker"
"I have read the license":
  type: "checkbox"
"Intended usage":
  type: "select"
   options:
      - label "Academic Research"
        value: "academic"
      - label: "Commercial & For-profit"
        value: "commercial"
      - label: "Personal exclusively"
        value: "personal"
```

Reply:
no strong opinion (given it's maybe a bit more complex to write it that way)

Reply:
ok for me to keep as is